src/main/java/se/iths/crimedatabase/security/SecurityConfig.java

```diff
@@ -0,0 +1,55 @@
+package se.iths.crimedatabase.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private UserPrincipalDetailsService userPrincipalDetailsService;
+
+    public SecurityConfig(UserPrincipalDetailsService userPrincipalDetailsService) {
+        this.userPrincipalDetailsService = userPrincipalDetailsService;
+    }
+
+    //Defines datasource for our users
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) {
+        auth.authenticationProvider(authenticationProvider());
+    }
+
+
+    //Used to authorize requests
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/criminals").hasRole("ADMIN") //Only admin can access criminals
+                .antMatchers("/victims").hasRole("ADMIN") //Only admin can access victims
+                .antMatchers("/users").hasRole("ADMIN") //Only admin can access users
+                .anyRequest().authenticated() //Authenticated users are authorized to make any request except the above.
+                .and()
+                .httpBasic(); //Uses http basic as authentication
+    }
+
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    }
+
+    @Bean
+    public DaoAuthenticationProvider authenticationProvider() {
+        DaoAuthenticationProvider provider = new DaoAuthenticationProvider();
+        provider.setPasswordEncoder(passwordEncoder());
+        provider.setUserDetailsService(userPrincipalDetailsService);
+        return provider;
+    }
```

Comment:
```suggestion
```

Reply:
We do not need this it seems

Reply:
Good! I will do some tests and remove it if all looks good!

Reply:
Remember to remove both this and the next sections from the SecurityConfig file before testing :)